internal/venice-common/src/main/java/com/linkedin/venice/acl/handler/StoreAclHandler.java

```diff
@@ -251,74 +131,158 @@ public <ReqT, RespT> ServerCall.Listener<ReqT> interceptCall(
+            .accept(httpResponseStatusToGrpcStatus(httpResponseStatus, s), headers));
 
-        if (VeniceSystemStoreUtils.isSystemStore(storeName)) {
-          super.onMessage(message);
+        if (StringUtils.isEmpty(storeName) || StringUtils.isEmpty(method)) {
+          LOGGER.error("Invalid store name {} or method {}", storeName, method);
+          grpcCloseConsumer.accept(Status.INVALID_ARGUMENT.withDescription("Invalid request"), headers);
           return;
         }
 
         try {
-          if (accessController.hasAccess(clientCert, storeName, method)) {
-            super.onMessage(message);
-          } else {
-            String client =
-                Objects.requireNonNull(call.getAttributes().get(Grpc.TRANSPORT_ATTR_REMOTE_ADDR)).toString();
-            String errLine = String.format("%s requested %s %s", client, method, storeName);
-
-            if (!accessController.isFailOpen() && !accessController.hasAcl(storeName)) {
-              LOGGER.warn("Requested store does not have ACL: {}", errLine);
-              LOGGER.debug(
-                  "Existing stores: {}",
-                  () -> metadataRepository.getAllStores()
-                      .stream()
-                      .map(Store::getName)
-                      .sorted()
-                      .collect(Collectors.toList()));
-              LOGGER.debug(
-                  "Access-controlled stores: {}",
-                  () -> accessController.getAccessControlledResources().stream().sorted().collect(Collectors.toList()));
-              String responseMessage = "ACL not found!\n" + "Either it has not been created, or can not be loaded.\n"
-                  + "Please create the ACL, or report the error if you know for sure that ACL exists for this store: "
-                  + storeName;
-              call.close(Status.PERMISSION_DENIED.withDescription(responseMessage), new Metadata());
-            } else {
-              String responseMessage = "Access denied!\n"
-                  + "If you are the store owner, add this application (or your own username for Venice shell client) to the store ACL.\n"
-                  + "Otherwise, ask the store owner for read permission.";
-              call.close(Status.PERMISSION_DENIED.withDescription(responseMessage), new Metadata());
-            }
-            return;
-          }
-        } catch (AclException e) {
+          X509Certificate clientCert = extractGrpcClientCert(call);
           String client = Objects.requireNonNull(call.getAttributes().get(Grpc.TRANSPORT_ATTR_REMOTE_ADDR)).toString();
-          String errLine = String.format("%s requested %s %s", client, method, storeName);
 
-          if (accessController.isFailOpen()) {
-            LOGGER.warn("Exception occurred! Access granted: {} {}", errLine, e);
+          if (VeniceSystemStoreUtils.isSystemStore(storeName)
+              || hasAccess(client, "", clientCert, storeName, method, errorHandler)) {
```

Comment:
Looks like URI is only used for informative purpose but why are we passing empty string `""` for URI?

Reply:
We don't have the request in case of GRPC. I initially left it empty for this reason. However, maybe we can populate it with just the authority (host w/ port) although that isn't the same as URI we get on the other path with netty.

What do you think?

Reply:
I see. Yeah just the authority is good too, better than nothing.